include/subgroup/tile/impl/load_xe.hpp

```diff
@@ -398,9 +398,8 @@ tile_load(tile_t& tile, payload_t& payload) {
   static constexpr gpu_arch arch_tag = payload_t::arch_tag;
 
   using load_store_attr = load_store_attr_t<msg_type::block_1d, arch_tag>;
-  static constexpr uint32_t max_load_vec_len = std::min(
```

Comment:
I made another kind of fix for that, like
diff --git a/include/subgroup/tile/impl/load_xe.hpp b/include/subgroup/tile/impl/load_xe.hpp
index 0c9d32ca..32f1c791 100644
--- a/include/subgroup/tile/impl/load_xe.hpp
+++ b/include/subgroup/tile/impl/load_xe.hpp
@@ -22,6 +22,7 @@
 #include <subgroup/tile/api.hpp>
 #include <subgroup/tile/impl/op_function.hpp>
 #include <subgroup/tile/impl/payload_xe.hpp>
+#include <bit>

 namespace gpu::xetla::subgroup {

@@ -395,12 +396,11 @@ __XETLA_API typename std::enable_if_t<
 tile_load(tile_t& tile, payload_t& payload) {
   using dtype = typename payload_t::dtype;
   static constexpr uint32_t load_len = tile_t::tile_elems;
+  static constexpr uint32_t power2_block_bytes = std::bit_ceil(tile_t::block_elems) * sizeof(dtype);
   static constexpr gpu_arch arch_tag = payload_t::arch_tag;

   using load_store_attr = load_store_attr_t<msg_type::block_1d, arch_tag>;
-  static constexpr uint32_t max_load_vec_len = std::min(
-      uint32_t(tile_t::block_elems * sizeof(dtype)),
-      load_store_attr::max_load_vec_len);
+  static constexpr uint32_t max_load_vec_len = std::min(power2_block_bytes, load_store_attr::max_load_vec_len);

   static constexpr uint32_t max_load_vec_elems =
       max_load_vec_len / sizeof(dtype);

could have a try whether this can keep the performance  ?